build-tools/xaprepare/xaprepare/Steps/Step_Android_SDK_NDK.cs

```diff
@@ -129,8 +129,12 @@ void CheckPackageStatus (Context context, string packageCacheDir, AndroidPackage
 			Log.StatusLine ($"  {context.Characters.Bullet} Installing ", pkg.Component.Name, tailColor: ConsoleColor.White);
 
 			if (File.Exists (pkg.LocalPackagePath)) {
-				if ((RefreshSdk && !IsNdk (pkg.Component)) || (RefreshNdk && IsNdk (pkg.Component))) {
-					LogStatus ("Reinstall requested, deleting cache", 4, ConsoleColor.Magenta);
+				bool valid = Utilities.VerifyArchive (pkg.LocalPackagePath).Result;
```

Comment:
I think it's best to never use `.Result`, and instead use `GetAwaiter ().GetResult ()`.

Reply:
Or better yet, if the whole method is async just await: 
```suggestion
				bool valid = await Utilities.VerifyArchive (pkg.LocalPackagePath);
```

Reply:
It looks like this method is not async, but it could be? https://github.com/xamarin/xamarin-android/blob/9ec9b5105b16291a6bbdd9d48a6ee1702a390235/build-tools/xaprepare/xaprepare/Steps/Step_Android_SDK_NDK.cs#L127

Reply:
I don't want to make it async because it's used in a LINQ query